src/modules/report/collector.py

```diff
@@ -3,9 +3,20 @@
 from __main__ import config
 from src.core.events import handler
 from src.core.events.types import Event, Service, Vulnerability, HuntFinished, HuntStarted
+import threading
 
+
+global services_lock
+services_lock = threading.Lock()
+services_lock.acquire()
 services = list()
+services_lock.release()
+
+global vulnerabilities_lock
+vulnerabilities_lock = threading.Lock()
+vulnerabilities_lock.acquire()
 vulnerabilities = list()
+vulnerabilities_lock.release()
```

Comment:
Why do that? There is no multithreading when loading of the module.

Reply:
Just to clarify, I think @danielsagi means you don't need to grab & release the lock here, because the code is only called once as the module is loaded. I think it does make sense to have these locks though.

Reply:
Thanks for the comment Daniel.
I agree with some of your comment- I should review my own code a bit more before submitting a PR (and I would do it the next times!).
But I do prefer locking more variables even if I'm not 100% sure they should be locked Because two main reasons:
1) The fact I think I have fixed the bug, found these exact 2 lines that it coming from and the program runs 100/100 successfully does not mean it would keep acting well in a few weeks when I would add new algorithms that might trigger another multi threading bug..
2) I do prefer better locking too much and having Liz reviewing my work so I can fix it comparing to leaving my code vulnerable to more threading bugs on the future...  (It is easier for a reviewer to see an overuse of the locking than actually checking allover the code for missing locks). 
:-)

Reply:
I still don't think you should have to grab the locks here as the code is only run once when the module is loaded

Reply:
You don't need to grab the locks here as the code is only run once when the module is loaded. It doesn't do a lot of harm but it's simpler not to do it (and good practice not to grab a lock unless you really need it)